lib/ModelMixins/DiscretelyTimeVaryingMixin.ts

```diff
@@ -96,21 +96,24 @@ function DiscretelyTimeVaryingMixin<
         return undefined;
       }
 
-      const asJulian: AsJulian[] = [];
+      const asJulianMap: Map<string, AsJulian> = new Map();
       for (let i = 0; i < discreteTimes.length; i++) {
         const dt = discreteTimes[i];
         try {
           if (dt.time !== undefined) {
             const time = JulianDate.fromIso8601(dt.time);
-            asJulian.push({
+            const asJulian = {
               time,
               tag: dt.tag !== undefined ? dt.tag : dt.time
-            });
+            };
+            asJulianMap.set(asJulian.tag, asJulian);
           }
         } catch {}
       }
-      asJulian.sort((a, b) => JulianDate.compare(a.time, b.time));
-      return asJulian;
+
+      const asJulianArray = Array.from(asJulianMap.values());
+      asJulianArray.sort((a, b) => JulianDate.compare(a.time, b.time));
+      return asJulianArray;
```

Comment:
I'd rather not have this change in `DiscretelyTimeVaryingMixin` as it could have performance implications. 
I think `discreteTimes` should be implemented to be unique in https://github.com/TerriaJS/terriajs/blob/4f9a5f84aac1b73d0a29f125fcea10ee1f39a319/lib/ModelMixins/GeojsonMixin.ts#L518

Reply:
What do you mean by performance implications? For things that use `discreteTimesAsSortedJulianDates`? This is why I originally had it as a separate getter. I don't want to put it in GeoJsonMixin because otherwise everything that wants no duplicates (which is most things!) will have to reimplement it, so we might as put it in DiscretelyTimeVaryingMixin.

Reply:
I'm saying that `discreteTimes` should only ever return unique dates - which is the case for every other implementation of `discreteTimes`. 
So with this change, it is de-duplicating twice - which I think may cause some performance issues - as we can have datasest with millions of dates

Reply:
OH
OH sorry I have no idea what I was thinking thank you for being patient with my nonsense
I think I've fixed it now